Get Python LDFLAGS from 'pkg-config'.

The other approach is more complex and gets it wrong: it returns
"-L/path/to/python/lib/python-2.7.10/lib/python2.7/config -lpython -lwhatever";
since that config/ sub-directory contains libpython2.7.a, we end up
statically linking Python in Weechat's python.so, which we do not want.

--- weechat-1.3/configure.ac	2015-08-16 08:27:07.000000000 +0200
+++ weechat-1.3/configure.ac	2015-11-29 18:18:52.975197048 +0100
@@ -535,29 +535,7 @@ if test "x$enable_python" = "xyes" ; the
             if test -r "$PYTHON_INCLUDE/Python.h"; then
                 PYTHON_CFLAGS="-I$PYTHON_INCLUDE"
                 AC_MSG_RESULT(found)
-                PYTHON_LIB=`$PYTHON -c "import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_config_var('LIBPL'))"`
-                PYTHON_LFLAGS="-lpython$PYTHON_VERSION "`$PYTHON -c "import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_config_var('LIBS')+' '+distutils.sysconfig.get_config_var('SYSLIBS')+' '+distutils.sysconfig.get_config_var('LINKFORSHARED'))"`
-                AC_MSG_CHECKING(for Python library)
-                if test -r "$PYTHON_LIB/libpython$PYTHON_VERSION.so"; then
-                   PYTHON_LFLAGS="-L$PYTHON_LIB $PYTHON_LFLAGS"
-                   AC_MSG_RESULT(found)
-                elif test -r "$PYTHON_LIB/libpython$PYTHON_VERSION.a"; then
-                   PYTHON_LFLAGS="-L$PYTHON_LIB $PYTHON_LFLAGS"
-                   AC_MSG_RESULT(found)
-                elif test -r "$PYTHON_LIB/libpython$PYTHON_VERSION.dll.a"; then
-                   PYTHON_LFLAGS="-L$PYTHON_LIB $PYTHON_LFLAGS"
-                   AC_MSG_RESULT(found)
-                elif test -r "$PYTHON_SYSPREFIX/lib/libpython$PYTHON_VERSION.so"; then
-                   PYTHON_LFLAGS="-L$PYTHON_SYSPREFIX/lib/ $PYTHON_LFLAGS"
-                   AC_MSG_RESULT(found)
-                else
-                    AC_MSG_WARN([
-*** Python library couldn't be found on your system.
-*** Try to install it with your software package manager.
-*** WeeChat will be built without Python support.])
-                    enable_python="no"
-                    not_found="$not_found python"
-                fi
+                PYTHON_LFLAGS=`pkg-config python2 --libs`
             else
                 AC_MSG_WARN([
 *** Python header files couldn't be found on your system.
